enclave-tls/src/tls_wrappers/internal/etls_tls_wrapper_select.c

```diff
@@ -32,6 +33,23 @@ enclave_tls_err_t etls_tls_wrapper_select(etls_core_context_t *ctx,
 		if (type && strcmp(type, tls_wrappers_ctx[i]->opts->type))
 			continue;
 
+		if (tls_wrappers_ctx[i]->opts->flags) {
+                        bool sgx_supported = (is_sgx2_supported() || is_sgx1_supported());
+                        if (!sgx_supported) {
+                                ETLS_ERR("Selecting tls wrapper while lacking sgx \
+                                 capability\n");
+                                return -ENCLAVE_TLS_ERR_INIT;
+                        }
+                } else {
```

Comment:
TLS_WRAPPER_OPTS_FLAGS_SGX_ENCLAVE implies:
- platform supports SGX, and
- sgx driver is available

So the driver check logic should be in the same if condition.